### PR TITLE
🎉 New Source: Zapier Supported Storage [low-code CDK]

### PR DESCRIPTION
**Pull request by albert-marrero**
## What
Adding a Source Connector for Zapier

See issue here: https://github.com/airbytehq/connector-contest/issues/11

## How
Using the Storage by Zapier. 

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No breaking change

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by albert-marrero** [comment]
test

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/schemas/TODO.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# TODO: Define your stream schemas
+Your connector must describe the schema of each stream it can output using [JSONSchema](https://json-schema.org). 
+
+You can describe the schema of your streams using one `.json` file per stream.
+ 
+## Static schemas
+From the `zapier_supported_storage.yaml` configuration file, you read the `.json` files in the `schemas/` directory. You can refer to a schema in your configuration file using the `schema_loader` component's `file_path` field. For example:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/schemas/TODO.md:7]: Remove this file.

File: airbyte-integrations/connectors/source-zapier-supported-storage/integration_tests/catalog.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      "supported_sync_modes": ["full_refresh", "incremental"],
+      "source_defined_cursor": true,
+      "default_cursor_field": "column1",
+      "json_schema": {
+        "$schema": "http://json-schema.org/draft-07/schema#",
+        "type": "object",
+        "properties": {
+          "column1": {
+            "type": "string"
+          },
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/integration_tests/catalog.json:14]: Remove this file.

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: Zapier Supported Storage Spec
+  type: object
+  required:
+    - secret
+  additionalProperties: true
+  properties:
+    secret:
+      type: string
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/spec.yaml:11]: add title to your property.

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+documentationUrl: https://docsurl.com
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/spec.yaml:1]: update the doc to `https://docs.airbyte.com/integrations/sources/zapier-supported-storage`

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/zapier_supported_storage.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+definitions:
+  selector:
+    extractor:
+      field_pointer: []
+  requester:
+    url_base: "https://store.zapier.com"
+    http_method: "GET"
+#    authenticator:
+#      type: BearerAuthenticator
+#      api_token: "{{ config['api_key'] }}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/zapier_supported_storage.yaml:12]: what that?

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/zapier_supported_storage.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  zapier_supported_storage_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "zapier_supported_storage"
+      primary_key: "data"
+      path: "/api/records"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/zapier_supported_storage.yaml:31]: please move the preview `/api` to the base_url

File: docs/integrations/sources/zapier-supported-storage.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+* Click Generate a new token.
+* In the Token name field, type a name for the token to help you identify it.
+* Choose needed scopes \(API actions this token can perform - or permissions it has\). See here for more details on scopes.
+* Click Generate token.
+
+## Changelog
+
+| Version | Date       | Pull Request                                             | Subject                                                                 |
+|:--------|:-----------|:---------------------------------------------------------|                        |
+| 0.1.0   | 2022-10-25 |  | Initial release for Zapier Supported Storage|
```
Comment (marcosmarxm) [inline docs/integrations/sources/zapier-supported-storage.md:56]: Add the Pr number.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
A few comments.

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/schemas/TODO.md (hunk 1 above)
Comment (albert-marrero) [reply airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/schemas/TODO.md:7]: Done.

**Comment by albert-marrero** [comment]
@marcosmarxm sorry for the delay, it was pretty crazy with work and getting a home ready for a baby. I hope I did everything correctly.

File: docs/integrations/README.md [hunk 8]
```diff
@@ -191,6 +191,7 @@ For more information about the grading system, see [Product Release Stages](http
 | [Wrike](sources/wrike.md)                                                                   | Alpha                | No                  |
 | [YouTube Analytics](sources/youtube-analytics.md)                                           | Beta                 | Yes                 |
 | [Xkcd](sources/xkcd.md)                                                                     | Alpha                | No                  |
+| [Zapier Supported Storage](sources/zapier-supported-storage.md)                             | Beta                 | Yes                 |
```
Comment (marcosmarxm) [inline docs/integrations/README.md:194]:
```suggestion
| [Zapier Supported Storage](sources/zapier-supported-storage.md)                             | Alpha                 | No                 |
```

File: docs/integrations/sources/zapier-supported-storage.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
+# Typeform
```
Comment (marcosmarxm) [inline docs/integrations/sources/zapier-supported-storage.md:1]: Please update the documentation it is using the other connector, Typeform.

File: airbyte-integrations/connectors/source-zapier-supported-storage/integration_tests/sample_config.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+  "fix-me": "TODO"
+}
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/integration_tests/sample_config.json:3]: Please correct this file.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments @albert-marrero

**Comment by marcosmarxm** [comment]
@albert-marrero can you share the integration test results besides the read command?

**Comment by albert-marrero** [comment]
@GreeneWoods , can you share the integration test results besides the read command?

**Comment by GreeneWoods** [comment]
updated pending issues in md and configs.

all tests are being passed
image

File: airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/schemas/TODO.md [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+```
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zapier-supported-storage/source_zapier_supported_storage/schemas/TODO.md:8]: Please remove this file...

File: docs/integrations/sources/zapier-supported-storage.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+To get the API token for your application follow this [steps](https://developer.typeform.com/get-started/personal-access-token/)
+
+* Log in to your account at Typeform.
+* In the upper-right corner, in the drop-down menu next to your profile photo, click My Account.
+* In the left menu, click Personal tokens.
+* Click Generate a new token.
+* In the Token name field, type a name for the token to help you identify it.
+* Choose needed scopes \(API actions this token can perform - or permissions it has\). See here for more details on scopes.
+* Click Generate token.
```
Comment (marcosmarxm) [inline docs/integrations/sources/zapier-supported-storage.md:50]: Please update this document correctly.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
There are some requests not solved yet.

**Comment by albert-marrero** [comment]
@marcosmarxm sorry about that, my partner and I thought we did everything correctly.  Hopefully it is done correctly now.

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
/publish connector=connectors/source-zapier-supported-storage

**Review by marcosmarxm — APPROVED** [review]

**Comment by RealChrisSean** [comment]
Hello @albert-marrero! Congrats on having this merged. Can you please provide an email for us to contact you about payouts?

**Comment by albert-marrero** [comment]
Hello @RealChrisSean what is the best way to give you that info?

**Pull request merged**